diff --git a/lua/roslyn/sln/api.lua b/lua/roslyn/sln/api.lua
index e5c6ec1..831ee98 100644
--- a/lua/roslyn/sln/api.lua
+++ b/lua/roslyn/sln/api.lua
@@ -12,16 +12,16 @@ local function sln_match(line, target)
 
     if ext == "sln" then
         local id, name, path = line:match('Project%("{(.-)}"%).*= "(.-)", "(.-)", "{.-}"')
-        if id and name and path and path:match("%.csproj$") then
+        if id and name and path and path:match("%.%a+proj$") then
             return path
         end
     elseif ext == "slnx" then
         local path = line:match('<Project Path="([^"]+)"')
-        if path and path:match("%.csproj$") then
+        if path and path:match("%.%a+proj$") then
             return path
         end
     elseif ext == "slnf" then
-        return line:match('"(.*%.csproj)"')
+        return line:match('"(.*%.%a+proj)"')
     else
         error(string.format("Unknown extension `%s` for solution: `%s`", ext, target))
     end
diff --git a/lua/roslyn/sln/utils.lua b/lua/roslyn/sln/utils.lua
index c530afb..12a211b 100644
--- a/lua/roslyn/sln/utils.lua
+++ b/lua/roslyn/sln/utils.lua
@@ -119,7 +119,7 @@ end
 ---@return string?
 local function find_csproj_file(bufnr)
     return vim.fs.find(function(name)
-        return name:match("%.csproj$") ~= nil
+        return name:match("%.%a+proj$") ~= nil
     end, { upward = true, path = vim.api.nvim_buf_get_name(bufnr) })[1]
 end
 
